src/main/java/games/strategy/io/IoUtils.java

```diff
@@ -0,0 +1,120 @@
+package games.strategy.io;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * A collection of useful methods related to I/O.
+ */
+public final class IoUtils {
+  private IoUtils() {}
+
+  /**
+   * Invokes the specified consumer with an in-memory input stream that wraps the specified byte array.
+   *
+   * @param bytes The byte array from which to read.
+   * @param consumer The consumer that will accept the input stream containing {@code bytes}.
+   *
+   * @throws IOException If {@code consumer} encounters an error while reading from the input stream.
+   */
+  public static void consumeFromMemory(final byte[] bytes, final InputStreamConsumer consumer) throws IOException {
```

Comment:
There's probably a better name for this method.  It was necessary to name it differently from `readFromMemory()` because the compiler can't distinguish between `InputStreamConsumer` and `InputStreamFunction` when lambdas are used at the call site.  Suggestions welcome.